packages/components/src/List/Toolbar/ColumnChooserButton/ColumnChooser/ColumnChooser.component.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { ColumnChooserProvider } from './columnChooser.context';
+import { useColumnChooserManager } from '../hooks';
+import FilterBar from '../../../../FilterBar';
+import ColumnChooserBody from './ColumnChooserBody';
+import ColumnChooserFooter from './ColumnChooserFooter';
+import ColumnChooserHeader from './ColumnChooserHeader';
+import getDefaultT from '../../../../translate';
+import cssModule from './ColumnChooser.scss';
+import { getTheme } from '../../../../theme';
+
+const theme = getTheme(cssModule);
+
+const haveColumnLabel = label => column => column.label.toLowerCase().includes(label.toLowerCase());
+const filterColumnsChooser = (columns, filter) => columns.filter(haveColumnLabel(filter));
+const changeVisibleToHidden = column => ({ ...column, hidden: !column.visible });
+const transformColumnsChooser = columns => columns.map(changeVisibleToHidden);
+
+export default function ColumnChooser({
+	children,
+	columns,
+	filterValue,
+	id,
+	nbLockedLeftItems = 0,
+	onClose,
+	submit,
+	t = getDefaultT(),
+}) {
+	const { columnsChooser, onChangeVisibility, onSelectAll, selectAll } = useColumnChooserManager(
```

Comment:
Are `columnsChooser ` the `columns` ? Something calls `columnsChooser` is disturbing to indicate a list of column.

Reply:
Waiting for a clear proposition of the naming you want. No change before that.

Reply:
columnsChooser will become columns in the component.
What about the props name columns in ColumnChooserButton and ColumnsChooser that represent the columns from the list ?